code/game/objects/items/weapons/shields.dm

```diff
@@ -150,8 +154,12 @@
 		spark(user.loc, 5)
 		playsound(user.loc, 'sound/weapons/blade.ogg', 50, 1)
 
+	var/shield_dir = user.dir
+	if(on_back)
+		shield_dir = reverse_dir[user.dir]
+
 	//block as long as they are not directly behind us
-	var/bad_arc = reverse_direction(user.dir) //arc of directions from which we cannot block
+	var/bad_arc = reverse_direction(shield_dir) //arc of directions from which we cannot block
```

Comment:
We're using both reverse_dir and reverse_direction. I imagine they do the same thing? Maybe you can pick one and see if the other can be removed.